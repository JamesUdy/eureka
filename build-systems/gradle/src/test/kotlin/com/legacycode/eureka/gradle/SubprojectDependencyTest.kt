package com.legacycode.eureka.gradle

import com.google.common.truth.Truth.assertThat
import com.legacycode.eureka.gradle.testing.TaskOutputResource
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class SubprojectDependencyTest {
  @ParameterizedTest
  @ValueSource(
    strings = [
      "+--- project libsignal-service",
      "+--- project :libsignal-service",
      "+--- project libsignal-service (n)",
      "+--- project libsignal-service (c)",
      "+--- project libsignal-service (*)",
      "+--- project :libsignal-service FAILED",
    ]
  )
  fun `it can parse variations of the subproject dependency patterns`(
    subprojectDependencyLine: String,
  ) {
    // when
    val dependency = SubprojectDependency.from(subprojectDependencyLine)!!

    // then
    assertThat(dependency.name)
      .isEqualTo("libsignal-service")
  }

  @Test
  fun `it does not parse variation that are not on level 1`() {
    // given
    val subprojectDependencyLine = "|    +--- project :core-util (*)"

    // when
    val dependency = SubprojectDependency.from(subprojectDependencyLine)

    // then
    assertThat(dependency)
      .isNull()
  }

  @Test
  fun `it can parse project dependencies from task output`() {
    // given
    val dependenciesOutput = TaskOutputResource.dependencies("Signal-Android.txt").output

    // when
    val subprojectDependencies = SubprojectDependency.from(dependenciesOutput)

    // then
    val subprojectDependenciesList = subprojectDependencies.joinToString("\n") { it.name }
    Approvals.verify(subprojectDependenciesList)
  }
}
