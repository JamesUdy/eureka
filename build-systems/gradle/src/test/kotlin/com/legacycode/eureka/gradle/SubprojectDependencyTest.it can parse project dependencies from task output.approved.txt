libsignal-service
paging
core-util
glide-config
video
device-transfer
image-editor
donations
contacts
qr
sms-exporter
sticky-header-grid
photoview
core-ui
Signal-Android
spinner
